### PR TITLE
Fix ftme__sell e2e spec

### PR DESCRIPTION
**Pull request by youbrokesomething**
## Proposed Changes

* Fix for ftme__sell spec that fails due to incorrect button text. Changes include a new step to click on a specific button locator.
  - The logic for the theme picker in the start site flow is: themes with style variations have a 'Continue' button, and themes without style variations have 'Start with {themeName}.' The theme currently used in the spec has style variations.
  - The theme preview screen can have one of two different continue buttons depending on the viewport size, so two new selectors were added. 

## Testing Instructions

* The ftme__sell e2e spec should pass.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by youbrokesomething** [comment]
I have just the two continue button selectors currently. I could add the 'Start with ${ themeName}' option to the race condition to allow the test to continue passing if we change the theme to a theme that doesn't have style variations.

Thoughts/opinions?

File: packages/calypso-e2e/src/lib/flows/start-site-flow.ts [hunk 1]
```diff
@@ -167,4 +169,16 @@ export class StartSiteFlow {
 	async selectTheme( themeName: string ): Promise< void > {
 		await this.page.getByRole( 'button', { name: themeName } ).click();
 	}
+
+	/**
+	 * Confirms Theme selection by clicking the correct Continue button depending on viewport size.
+	 * themeContinueButton will be visible at viewports above 1080px.
+	 * themeContinueButtonSmall will be visible at viewports below 1080px.
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/flows/start-site-flow.ts:176]: **comment:blocking** - turns out, the assumption here is not entirely correct!

As you can see here as an example, the Winkel theme shows the same `Continue` button regardless of the horizontal viewport width.

Desktop
image

Mobile
image
----

Instead, the actual difference is whether the theme is FSE-enabled or not. You can see the difference with Vivre below, which is **not** a FSE theme.
image

So in summary, we have two situations:
- theme is FSE: `Continue` button is shown to confirm the selected theme.
- theme is not FSE: `Start with ${themeName}` is button is shown to confirm the selected theme.

### So what does this mean?

This has implications for the added selectors in `StartSiteFlow` - we don't actually need to use any new selectors!

File: packages/calypso-e2e/src/lib/flows/start-site-flow.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ const selectors = {
 
 	// Themes
 	individualThemeContainer: ( name: string ) => `.design-button-container:has-text("${ name }")`,
+	themeContinueButton: '.design-preview__sidebar-action-buttons:has-text("Continue")',
+	themeContinueButtonSmall: '.step-container__navigation:has-text("Continue")',
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/flows/start-site-flow.ts:25]: **issue:blocking** - because the distinction is between FSE and non-FSE themes, this selector is not actually going to work if we opt to test a theme that is not FSE. That's because for such themes, the button to confirm theme selection reads "Start with ${themeName}` instead.

I should have been clearer that we do not need to worry about mobile viewports for the time being.

Comment (worldomonation) [reply packages/calypso-e2e/src/lib/flows/start-site-flow.ts:25]: **issue:continuing** - Furthermore with Playwright adding support for obtaining locators by one of the many a11y attributes, it would be preferable to begin deprecating the use of selector objects as time goes on. As an added bonus, it makes the code easier to follow.

An example of what Playwright can do:

```typescript
await this.page.getByRole( 'button', { name: "Continue" } ).click();
```

**Review by worldomonation — CHANGES_REQUESTED** [review]
The structure of the changes are good - adding an explicit step to confirm the theme selection is a good idea! 

That said, the initial assumption we worked from turned out to be unfortunately incorrect, so let's pivot to not worry about Mobile viewport for the time being and support both FSE and non-FSE theme selections.

**Comment by worldomonation** [comment]
Also, **issue:blocking** - please update the title to follow the convention used by KitKat.

**Comment by worldomonation** [comment]
@youbrokesomething Checking up on the progress of this PR. If you would like some guidance and/or for me to take over the PR, let me know.

**Comment by worldomonation** [comment]
Superseded by https://github.com/Automattic/wp-calypso/pull/76659.
